freqtrade/freqai/prediction_models/ReinforcementLearner_multiproc.py

```diff
@@ -0,0 +1,88 @@
+import logging
+from pathlib import Path
+from typing import Any, Dict  # , Tuple
+
+# import numpy.typing as npt
+import torch as th
+from pandas import DataFrame
+from stable_baselines3.common.callbacks import EvalCallback
+from stable_baselines3.common.vec_env import SubprocVecEnv
+
+from freqtrade.freqai.data_kitchen import FreqaiDataKitchen
+from freqtrade.freqai.RL.BaseReinforcementLearningModel import (BaseReinforcementLearningModel,
+                                                                make_env)
+
+
+logger = logging.getLogger(__name__)
+
+
+class ReinforcementLearner_multiproc(BaseReinforcementLearningModel):
+    """
+    User created Reinforcement Learning Model prediction model.
+    """
+
+    def fit(self, data_dictionary: Dict[str, Any], dk: FreqaiDataKitchen, **kwargs):
+
+        train_df = data_dictionary["train_features"]
+        total_timesteps = self.freqai_info["rl_config"]["train_cycles"] * len(train_df)
+
+        # model arch
+        policy_kwargs = dict(activation_fn=th.nn.ReLU,
+                             net_arch=self.net_arch)
+
+        if dk.pair not in self.dd.model_dictionary or not self.continual_learning:
+            model = self.MODELCLASS(self.policy_type, self.train_env, policy_kwargs=policy_kwargs,
+                                    tensorboard_log=Path(
+                                        dk.full_path / "tensorboard" / dk.pair.split('/')[0]),
+                                    **self.freqai_info['model_training_parameters']
+                                    )
+        else:
+            logger.info('Continual learning activated - starting training from previously '
```

Comment:
Wouldn't this whole class benefit from subclassing `ReinforcementLearner`?

the fit method is identical apart from this comment (which says learning here, and training in `ReinforcementLearner`) - and could therefore then be removed.

Having this as a subclass would make `ReinforcementLearner_multiproc` runnable out of the box - currently it's not as it's missing a valid `MyRLEnv` implementation.

Reply:
yes, I like that idea. I changed it. And I improved docstring in `ReinforcementLearner` to help users understand how to inherit from it and create their own custom classes.